<launch>

    <node name="sensortf" pkg="tf" type="static_transform_publisher" args="0.539 0 0.75 1.57 0 1.309 girona1000/base_link girona1000/laser_link 100"/>

    <!-- <node name="sensortf" pkg="tf" type="static_transform_publisher" args="0.60 0 0.90 1.57 0 1.57 girona1000/base_link girona1000/laser_link 100"/> -->
    <!-- <node name="sensortf" pkg="tf" type="static_transform_publisher" args="0.60 0 0.90 1.57 0 1.57 girona1000/base_link girona1000/laser_link 100"/> -->

    <!-- <node pkg="hrov_martech2023" type="laser2ptcloud" name="laser2ptcloud" output="screen">
        <param name="laser_topic" value="girona1000/profiler"/>
    </node> -->

    <!-- depth image to pointcloud nodes -->
    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
    <node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager">
        <remap from="camera_info" to="/girona1000/depth/camera_info"/>
        <remap from="image_rect" to="/girona1000/depth/image_depth"/>
        <remap from="points" to="/girona1000/depth/points"/>
    </node>

    <!-- filtering pointcloud to octomap nodes -->
    <node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid nodelet_manager" output="screen">
        <remap from="~input" to="/girona1000/depth/points"/>
        <rosparam>
                filter_field_name: z
                filter_limit_min: 0.01
                filter_limit_max: 10
                filter_limit_negative: False
                leaf_size: 0.02
            </rosparam>
    </node>


    <!-- <group ns="dummy"> -->
    <param name="~frame_id" type="string" value="world"/>
    <param name="~base_frame_id" type="string" value="girona1000/base_link"/>
    <node pkg="octomap_server" type="octomap_server_node" name="octomap" output="screen">
        <!-- <remap from="/cloud_in" to="/girona1000/depth/points" /> -->
        <!-- <remap from="/cloud_in" to="/my_cloud"/> -->
        <remap from="/cloud_in" to="/voxel_grid/output"/>
        <param name="resolution" value="0.1"/>
        <!-- <param name="pointcloud_min" value="0.2"/>
        <param name="pointcloud_max" value="3"/>
        <param name="occupancy_max" value="-1"/> -->
        <!-- <param name="latch" value="false"/> -->
        <!-- <param name="publish_free_space" value="true"/> -->
        <!-- <param name="sensor_model/max_range" value="5.0"/> -->
        <param name="color/b" value="1"/>
        <param name="height_map" value="false"/>

    </node>
    <!-- </group> -->
</launch>